### ISSUE TITLE
Parse password-encrypted excel files

### ISSUE THREAD

**Issue by ryan-preble** [nushell/nushell#10546]
### Related problem

Excel gives you the option to encrypt spreadsheets. In the GUI, you would see this as a blank screen until you enter the password, and then excel will parse the file and display it. This might be neat to do from the command line too. Currently, nushell does not capture the password request in any way and just attempts to parse the encrypted contents.

![image](https://github.com/nushell/nushell/assets/70269632/aa4effdc-26b3-4b1a-ad85-93a3a07b6cf9)

### Describe the solution you'd like

It would be pretty cool if the user was prompted for a password when they try to exectute a command that opens an encrypted excel file. 

### Describe alternatives you've considered

Never use encrypted spreadsheets

### Additional context and details

Normally .docx and .xlsx files are actually just zipped folders, but that does not appear to be the case with encrypted excel files. If you change the file extension to .zip, it cannot be extracted, so I think the entire zipped folder rather than just the spreadsheet contents are encrypted. This is the raw file output from an encrypted excel file, which maybe you can use to detect if the user needs to be prompted for a password:
![image](https://github.com/nushell/nushell/assets/70269632/4531f053-49b8-4e25-92f6-2d8c408b1bac)

**Comment by amtoine** [comment]
@ryan-preble 

i'm curious here :smirk: 

with such a file, is there some command that you can run before to decrypt the file and then open it as a normal excel file?

i know GPG and i would 
- decrypt with GPG
- then open with `open`

**Comment by fdncred** [comment]
@amtoine I don't believe so. It's encrypted for a reason, to force someone to put a password in within Excel. IIRC, there are several types of encryption too. Providing a password on the cli may be an option if our crate that opens xlsx files supports it. Anything else would be hackery that I wouldn't want to support. I don't want nushell to be known as a cracking tool.

**Comment by ryan-preble** [comment]
> @ryan-preble
> 
> i'm curious here 😏
> 
> with such a file, is there some command that you can run before to decrypt the file and then open it as a normal excel file?
> 
> i know GPG and i would
> 
> * decrypt with GPG
> * then open with `open`

If there is a way, I can't figure it out. The windows cipher tool does not appear to be involved in the encryption of the file, as it will show the encrypted spreadsheet as being unencrypted. GPG gives me an error `gpg: [don't know]: invalid packet (ctb=43)` that only leads to unanswered twenty year old discussion boards. I think the encryption procedure excel uses stands alone. But you do bring up a (perhaps better) alternative where the file is encrypted using another tool just like any other file.

**Comment by sholderbach** [comment]
In [calamine](https://github.com/tafia/calamine) which we use for spreadsheet handling I can only find an error associated with password-protected xls files: https://docs.rs/calamine/latest/calamine/enum.XlsError.html#variant.Password

Open issue for XLSX https://github.com/tafia/calamine/issues/102
